src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if ([sideA, sideB, sideC].some((side) => side === 0)) {
+      throw new Error('Side is less than 0');
+    }
+
+    if (sideA >= sideB + sideC
+      || sideB >= sideA + sideC
+      || sideC >= sideA + sideB) {
+      throw new Error('One side is longer than sum of two others');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.sideA + this.sideB + this.sideC) / 2;
+
+    return +Math.sqrt(semiPerimeter * (semiPerimeter - this.sideA)
+    * (semiPerimeter - this.sideB) * (semiPerimeter - this.sideC)).toFixed(2);
```

Comment:
Optional: you could round with the same logic in 3 functions over the code. You also could create a separate helper function - `rounder` for example for this purpose. (use `toFixed(2) or Math.floor like in line 60`)